### ISSUE TITLE
[Simple Payments] Adding a Shipping & Billing address to a simple payment loses user entered info

### ISSUE THREAD

**Issue by joshheald** [woocommerce/woocommerce-android#5515]
**Describe the bug**
Typing out Shipping Address for a Simple Payment order, then turning on `Use as billing address` causes an error 

> Unable to update address with empty email. 
> Make sure you are running the latest version of Wo... 

The message is truncated on device

There is no email field on the `Shipping details` form to complete

The shipping details that were entered are lost

**To Reproduce**
1. Turn on Simple Payments
2. Go to orders
3. Tap `+`
4. Enter an amount, tap continue
5. Tap Shipping details and enter address details, complete all fields
6. Turn on the `Use as Billing Address` toggle
7. Tap Done

You will be returned to the order details screen and the shipping details will be blank with an address showing
Repeating with `Use as Billing Address` switched off, results in no error and the details being saved.

**Screenshots**

https://user-images.githubusercontent.com/2472348/147239791-ded84a2c-6299-41ab-b2ce-668f40015cc8.mp4

**Expected behavior**
The error is fine (though not being truncated would be nice), the issue is that the information the merchant entered is lost. It would be better if this was still saved, despite the error.

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: Emulator
- Android version: 30
- WooCommerce Android version: 8.2 beta

**Comment by nbradbury** [comment]
I moved this from "Simple Payments" to "Order Creation" because the bug is actually in the order detail screen. Simple payments used to redirect to the order detail screen, but now it has its own UI which doesn't include an address.

**Comment by nbradbury** [comment]
I'm closing this as I'm pretty sure it was fixed by [this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2307). The error message was due to the fact that WooCommerce <= v5.8.1 wouldn't allow an empty billing email, but it turned out if we passed the email as `null` rather than an empty string it worked.

I just downgrade WooCommerce on one of my sites to v5.7 and was unable to duplicate this issue.
